__device-tests__/gutenberg-editor-writing-flow.test.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+
+describe( 'Gutenberg Editor - Writing Flow', () => {
+	describe( 'TC007 - Test format detection under the cursor', () => {
+		it( 'checks that the proper format buttons are selected when the cursor is under', async () => {
+			const paragraphHTML = `<!-- wp:paragraph -->
+            <p><span accessibility-id="bold"><strong>bold</strong></span>
+            <span accessibility-id="bold-italic"><strong><i>bold-italic</i></strong>.</span>
+            <span accessibility-id-"strikethrough"><s>strikethrough</s></span></p>
+            <!-- /wp:paragraph -->`;
+
+			// On a rich-text based component, add bold, italic, strikethrough and link formatted text
+			await editorPage.addNewBlock( blockNames.paragraph );
+			const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.paragraph
+			);
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphHTML
```

Comment:
Thanks for outlining this. I've [updated the code](https://github.com/wordpress-mobile/gutenberg-mobile/blob/7e7576602fc06c4baa3a49140550b158c2311fbd/__device-tests__/gutenberg-editor-writing-flow.test.js#L10-L51) for option (B), to type plaintext and interact with the formatting buttons. 

With this approach, do you feel like it would make sense to take a visual snapshot at each step? E.g.:

```
await editorPage.typeTextToTextBlock(
	paragraphBlockElement,
        'text'
);

// Toggle various formatting options, then proceed with typing more text
await editorPage.toggleFormatting( 'Bold' );

screenshot = await takeScreenshot();
expect( screenshot ).toMatchImageSnapshot();

// ...repeat with taking a screenshot and an assertion
// for each of the remaining formatting options [?]
```